chef/automate
# Desktop View: Change Tracking: Create and List Rollouts

### :nut_and_bolt: Description: What code changed, and why?

Implements create/read/list for rollout objects. Rollout objects represent the user's intent to have some set of Chef Infra nodes run a certain version of Chef Code. Rollouts contain some data describing which nodes are supposed to run the code, based on the policy name, policy group, and Chef Server URL. Rollouts also contain some data describing the code, such as the policyfile revision_id, some optional git/github info, and some optional Ci info.

Future work will associate rollouts to Chef Client runs so we can show users the status of their rollouts based on how many nodes have run the latest codes and the client run status outcomes.

### :chains: Related Resources

* #3568
* #3222

### :+1: Definition of Done

create/read/list the rollout objects. This PR is already large so I'm cutting it off.

### :athletic_shoe: How to Build and Test the Change

* build automate-gateway and config management service
* start all the services
* `config_mgmt_list_rollouts` to list the rollouts (initially you should have zero)
* `config_mgmt_create_rollout` to create a rollout
* The studio helpers are designed for piping so you can use `jq` to create more varied data
